library/Vanilla/Utility/ObjectUtils.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * @author Todd Burry <todd@vanillaforums.com>
+ * @copyright 2009-2020 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Utility;
+
+/**
+ * Static methods usefull for operating on objects.
+ */
+final class ObjectUtils {
+    /**
+     * Hydrate an object from an array.
+     *
+     * The array is parsed where keys represent properties to be set by the values. To set a property, the following is done:
+     *
+     * 1. Look for a method in the form: `set$property`.
+     * 2. Look for a method in the form: `with$property`.
+     * 3. Look for an actual property.
+     *
+     * @param object $object The object to hydrate.
+     * @param array $data The data to hydrate with.
+     * @return object Returns the same class or a new one in the case of with methods.
```

Comment:
Since we're always modifying the actual `$object` instance, what's the intention of returning a value too? A fluent interface?

Reply:
I was thinking of this:

```php
$o = ObjectUtils::hydrate(new Something(), ['foo' => 'bar', ...]);
```

It's also useful for regular fluent calls.